src/rt-app_parse_config.c

```diff
@@ -363,6 +365,44 @@ static char* create_unique_name(char *tmp, int size, const char* ref, long tag)
 	return tmp;
 }
 
+void initialise_external_library(rtapp_resource_t *library_struct)
+{
+	/* dlopen(char *filename, int flag) */
+	void *library_handle;
+	void *symbol;
+	char const* shared_library_dir_name = get_shared_library_base_path();
+	if (!shared_library_dir_name)
+	{
+		log_error("failed to obtain shared library directory\n");
```

Comment:
exit(EXIT_FAILURE)?
Also please don't mix declarations and code.

Reply:
done